src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+            readDataArray[idx].insertionGOP = ReadUtils.getBaseInsertionQualities(read);
+            readDataArray[idx].deletionGOP = ReadUtils.getBaseDeletionQualities(read);
+            readDataArray[idx].overallGCP = gcp.get(read);
+            ++idx;
+        }
+
+        mLogLikelihoodArray = new double[readListSize * numHaplotypes];      //to store results
+        if (doProfiling)
+            threadLocalSetupTimeDiff = (System.nanoTime() - startTime);
+        //for(reads)
+        //   for(haplotypes)
+        //       compute_full_prob()
+        jniComputeLikelihoods(readListSize, numHaplotypes, readDataArray, mHaplotypeDataArray, mLogLikelihoodArray, 12);
+
+        int readIdx = 0;
+        for (int r = 0; r < readListSize; r++) {
+            int hapIdx = 0;
+            for (final Haplotype haplotype : logLikelihoods.alleles()) {
+
+                //Since the order of haplotypes in the List<Haplotype> and alleleHaplotypeMap is different,
+                //get idx of current haplotype in the list and use this idx to get the right likelihoodValue
+                final int idxInsideHaplotypeList = haplotypeToHaplotypeListIdxMap.get(haplotype);
+                logLikelihoods.set(hapIdx, r, mLogLikelihoodArray[readIdx + idxInsideHaplotypeList]);
+                ++hapIdx;
+            }
+            readIdx += numHaplotypes;
+        }
+        if (doProfiling) {
+            threadLocalPairHMMComputeTimeDiff = (System.nanoTime() - startTime);
+            //synchronized(doProfiling)
+            {
+                pairHMMComputeTime += threadLocalPairHMMComputeTimeDiff;
+                pairHMMSetupTime += threadLocalSetupTimeDiff;
+            }
+        }
+    }
+
+    /**
+     * Print final profiling information from native code
+     */
+    public native void jniClose();
+
+    @Override
+    public void close() {
+        if (doProfiling)
+            logger.info("Time spent in setup for JNI call : " + (pairHMMSetupTime * 1e-9));
+        super.close();
+        jniClose();
+    }
+
+    //Copied from http://frommyplayground.com/how-to-load-native-jni-library-from-jar 
+
+    /**
+     * Loads library from current JAR archive
+     * <p/>
+     * The file from JAR is copied into system temporary directory and then loaded. The temporary file is deleted after exiting.
+     * Method uses String as filename because the pathname is "abstract", not system-dependent.
+     *
+     * @param path The filename inside JAR as absolute path (beginning with '/'), e.g. /package/File.ext
+     * @throws IOException  If temporary file creation or read/write operation fails
+     * @throws IllegalArgumentException If source file (param path) does not exist
+     * @throws IllegalArgumentException If the path is not absolute or if the filename is shorter than three characters (restriction of {@see File#createTempFile(java.lang.String, java.lang.String)}).
+     */
+    public static void loadLibraryFromJar(String path) throws IOException {
```

Comment:
I think most of this method can be replace with a call to `IOUtils.writeTempResource(new Resource(path,  VectorLoglessPairHMM.class)`